packages/framework/src/Model/Category/AutomatedFilter/SpecialPricesCategoryAutomatedFilter.php

```diff
@@ -13,9 +14,11 @@ class SpecialPricesCategoryAutomatedFilter implements CategoryAutomatedFilterInt
 
     /**
      * @param \Shopsys\FrameworkBundle\Model\Customer\User\CurrentCustomerUser $currentCustomerUser
+     * @param \Shopsys\FrameworkBundle\Model\Pricing\PricingSetting $pricingSetting
      */
     public function __construct(
         protected readonly CurrentCustomerUser $currentCustomerUser,
+        protected readonly PricingSetting $pricingSetting,
```

Comment:
Instead of passing the pricing type in all the methods, you can rather pass the type via constructor in `FilterQueryFactory::create()`, that should simplify it quite a lot